improve typescript, added pnpm instead of yarn, updated dependencies, improved typescript config strictness

JustFly1984 opened the pull request:
@dvlkv Hello, please review. I found some issues - and fixed while improving the typescript config. Tried to impact as less as possible.

improve typescript, added pnpm instead of yarn, updated dependencies, improved typescript config strictness

Please review.

JustFly1984 commented:
@dvlkv  please review

dvlkv commented:
Hi! I am afraid I cannot merge this PR as it is affecting too much files and brings no practical value. Thank you for your contribution.

JustFly1984 commented:
It dos brings value: latest typescript, I've fixed some undefined or null type errors, updated types to support tsconfig settings. How is that not valuable?
If you not merging it, I will consider to just inline the code from this package into my own project, cos we using typescript 5.6, and you's stuck on typescript 4.
